tools/data-handler/src/utils/user-preferences.ts

```
@@ -95,17 +95,23 @@ export class UserPreferences {
   };
 
   constructor(private prefsFilePath: string) {
-    if (!existsSync(this.prefsFilePath)) {
-      // Create the preferences directory based on prefsFilePath dirname
-      const prefsDir = dirname(this.prefsFilePath);
+    // Create the preferences directory based on prefsFilePath dirname
+    const prefsDir = dirname(this.prefsFilePath);
 
-      try {
-        if (!existsSync(prefsDir)) {
-          mkdirSync(prefsDir, { recursive: true });
-        }
-        // File does not exist, create it with default content
-        writeFileSync(this.prefsFilePath, formatJson(UserPreferences.defaults));
-      } catch (error) {
+    try {
+      // Ensure directory exists first
+      if (!existsSync(prefsDir)) {
+        mkdirSync(prefsDir, { recursive: true });
+      }
+
+      // Try to create file exclusively using 'wx' flag
+      // This will fail atomically if file already exists
+      writeFileSync(this.prefsFilePath, formatJson(UserPreferences.defaults), {
+        flag: 'wx',
+      });
+    } catch (error: any) {
+      // If file already exists (EEXIST), that's fine - we'll use the existing file
+      if (error.code !== 'EEXIST') {
```

Could put a `logger.warn` statement in else. I think it's good practice when ignoring exceptions

---

Also I think you could use `error?.code` so it won't fail if error does not have `code`